src/abstractdataframe/abstractdataframe.jl

```diff
@@ -380,57 +380,69 @@ end
 ```
 
 **Arguments**
 
 * `df` : the AbstractDataFrame
-* `io` : optional output descriptor
-* `stats::AbstractVector{Symbol}`: the summary statistics to report. Allowed 
-  fields are `:mean`, `:std`, `:min`, `:q25`, `:median`, `:q75`, `:max`, `:eltype`, 
-  `:nunique`, and `:nmissing`
+* `stats::Union{Symbol,AbstractVector{Symbol}}` : the summary statistics to report. If 
+  a vector, allowed fields are `:mean`, `:std`, `:min`, `:q25`, `:median`, 
+  `:q75`, `:max`, `:eltype`, `:nunique`, `:first`, `:last`, and `:nmissing`. If set to
+  `:all`, all summary statistics are reported.
 
 **Result**
 
 * A `DataFrame` where each row represents a variable and each column a summary statistic.
 
 **Details**
 
-If the column's base type derives from `Number`, compute the mean, standard 
-deviation, minimum, first quantile, median, third quantile, and maximum. If 
-a column is not numeric, these statistics are populated with `nothing`s. 
+For `Real` columns, compute the mean, standard deviation, minimum, first quantile, median, 
+third quantile, and maximum. If a column does not derive from `Real`, `describe` will 
+attempt to calculate all statistics, using `nothing` as a fall-back in the case of an error.
 
 When `stats` contains `:nunique`, `describe` will report the 
-number of unique values in a column. If a column's base type derives from `Number`,
+number of unique values in a column. If a column's base type derives from `Real`,
 `:nunique` will return `nothing`s.
 
 Missing values are filtered in the calculation of all statistics, however the column
 `:nmissing` will report the number of missing values of that variable. 
 If the column does not allow missing values, `nothing` is returned. 
 Consequently, `nmissing = 0` indicates that the column allows
-missing values, but does not currently contain any. 
+missing values, but does not currently contain any.
 
 **Examples**
 
 ```julia
 df = DataFrame(i = 1:10, x = rand(10), y = rand(["a", "b", "c"], 10))
 describe(df)
+describe(df, stats = :all)
+describe(df, stats = [:min, :max])
 ```
 
 """
-StatsBase.describe(df::AbstractDataFrame; kwargs...) = describe(stdout, df; kwargs...)
-function StatsBase.describe(io::IO, df::AbstractDataFrame; stats::AbstractVector{Symbol} = 
-                            [:mean, :min, :median, :max, :nmissing, :eltype])
+function StatsBase.describe(df::AbstractDataFrame; stats::Union{Symbol,AbstractVector{Symbol}} = 
+                            [:mean, :min, :median, :max, :nunique, :nmissing, :eltype])
     # Check that people don't specify the wrong fields. 
     allowed_fields = [:mean, :std, :min, :q25, :median, :q75, 
-                      :max, :nunique, :nmissing, :eltype] 
-    if !issubset(stats, allowed_fields) 
+                      :max, :nunique, :nmissing, :first, :last, :eltype]
+    allowed_msg = "\nAllowed fields are: :" * join(allowed_fields, ", :")
```

Comment:
Better call this only in case of errors. It's not a big deal if you repeat this twice.